<launch>
  <!--
    'Y' button on the NYKO is the deadman switch
    Push the left frontal button labeled as 'LB' to activate cmd_vel publishing.
    Move the left stick around to control the velocity.
   -->

  <node pkg="car_teleop" type="car_teleop_joy" name="car_teleop_joystick">
    <param name="scale_angular" value="1.0"/>
    <param name="scale_linear" value="1.0"/>
    <!-- 'Y' button is the deadman switch 
         Button value = 5 
    -->
    <param name="axis_deadman" value="5"/>
    <!-- Throttle is the right trigger 
         Axis value = 4
    -->
    <param name="axis_linear" value="3"/>
    <param name="axis_angular" value="2"/>

    <param name="button_a" value="0"/>
    <param name="button_b" value="1"/>
    <param name="button_x" value="3"/>
    <!-- button y is mapped to the deadman -->
    <!-- buttons left and right are located above the triggers -->
    <param name="button_left" value="6"/>
    <param name="button_right" value="7"/>

  </node>

  <arg name="joy_dev" default="/dev/input/js0" />
  
  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="dev" value="$(arg joy_dev)" />
    <param name="deadzone" value="0.0" />
  </node>

</launch>
